src/Utility/ExprFunctions.h

```diff
@@ -42,23 +42,36 @@ class EvalFunc : public ScalarFunc
 
   std::vector<Real*> arg; //!< Function argument values
 
+  EvalFunc* gradient; //!< First derivative expression
+
+  Real dx; //!< Domain increment for calculation of numerical derivative
+
 public:
+  static int numError; //!< Error counter - set by the exception handler
+
   //! \brief The constructor parses the expression string.
-  EvalFunc(const char* function, const char* x = "x" );
+  EvalFunc(const char* function, const char* x = "x", Real eps = Real(1.0e-8));
   //! \brief The destructor frees the dynamically allocated objects.
   virtual ~EvalFunc();
 
-  static int numError; //!< Error counter - set by the exception handler
+  //! \brief Adds an expression function for a first derivative.
+  void derivative(const std::string& function, const char* x = "x");
+
+  //! \brief Returns whether the function is time-independent or not.
+  virtual bool isConstant() const { return false; }
+
+  //! \brief Returns the first-derivative of the function.
+  virtual Real deriv(Real x) const;
 
 protected:
   //! \brief Non-implemented copy constructor to disallow copying.
   EvalFunc(const EvalFunc&);
-  //! \brief Non-implemented assigment operator to disallow copying.
+  //! \brief Non-implemented assignment operator to disallow copying.
   EvalFunc& operator=(const EvalFunc&);
```

Comment:
c++11 has a useful construct - = delete. yields more understandable error messages should this be attempted to be used.

Reply:
changed.